benchmark/benchmark.py

```diff
@@ -167,12 +168,30 @@ def cli():
     # import the requested benchmark module
     mod = importlib.import_module(args.benchmark)
 
+    # check for additional device args
+    devs = copy(args.device)
+    dev_kwargs = []
+    for idx, dev in enumerate(devs):
+        dev_kwargs.append({})
+        if "default.tensor" in dev:
+            # check for additional args
+            device_full = dev.split("-")
+            devs[idx] = device_full[0]
+            if len(device_full) > 1:
+                dev_kwargs[idx]["representation"] = device_full[1]
```

Comment:
I think I follow what is happening here, but just to make I understand fully --- how are the new kwargs specified from the command line?

Like so?

```console
$ python3 bm_nearest_neighbour_circuit.py -d default.qubit,default.tensor-mps
```

It should be relatively simply to add a CLI argument to the benchmarking tool to accept optional device kwargs (and about the same amount of added code used here):

```python
import argparse

class ParseKwargs(argparse.Action):
    def __call__(self, parser, namespace, values, option_string=None):
        setattr(namespace, self.dest, dict())
        for value in values:
            key, value = value.split('=')
            getattr(namespace, self.dest)[key] = value

parser = argparse.ArgumentParser()
parser.add_argument('--dev-options', nargs='*', action=ParseKwargs)
args = parser.parse_args()
print(args.dev_options)
```

and then, down below:

```python
qml.device(short_name, wires=args.wires, **args.dev_options)
```

Reply:
Added to the list